charts/redpanda/values.yaml

```diff
@@ -102,25 +102,21 @@ tls:
 #
 # External access configuration
 external:
-  # Default external access value for all listeners except RPC
-  # External config doesn't apply to RPC listeners as they are never externally accessible
-  # These values can be overridden by each listener if needed
+  # Default external access value
+  # This be overridden by each listener if needed
   enabled: true
-  # Default external access type (only NodePort at the moment)
+  # External access type. `NodePort` is the only acceptable value if defined.
+  # If undefined, then advertised listeners will be configured and a service must be created manually
   type: NodePort
```

Comment:
That's a good point, it's difficult (impossible?) to "undefine" a parameter with helm set. I'll make this change.

Reply:
Actually I was thinking about another possibility rather than creating a `LoadBalancer` value for `external.type` within this PR.

Right now this PR has nothing tying it directly to a LoadBalancer service. It just makes it possible for the user to manually create the service that will provide external connectivity, whatever it is. I do think adding `LoadBalancer` to the helm chart is a good idea, as that will simplify creating a LoadBalancer service... but there are questions to answer and technical issues to solve.

One such issue is one vs many LoadBalancers. This PR makes it possible to define a different port for each node, which means that I think it could work to provide external access via a single LoadBalancer service (where different ports get you to different brokers). This would be a big benefit, but we need to make sure and handle the case where the user defines the same port for each node (which would only work with multiple LoadBalancers). We also need to make sure how it's defined in values.yaml is the most understandable way to do so.

Instead of answering these questions above here, I suggest we add in a `none` value for `external.type`. This would do what it suggests: no service of any type would be created. Instead you are only defining advertised listeners. Later once we have an actual LoadBalancer PR up and have answered the above questions, then we can add a `LoadBalancer` option.

Please post if you have any feedback, or take a look at the incoming (small) change to this PR.

Reply:
Sounds sensible, thanks @vuldin

Reply:
> That's a good point, it's difficult (impossible?) to "undefine" a parameter with helm set. I'll make this change.

```
--set things.unwanted=null
```

will remove "unwanted" from the "things" dict